packages/core/content-manager/server/src/controllers/collection-types.ts

```diff
@@ -417,14 +417,22 @@ export default {
 
       const isUpdate = !isCreate;
       if (isUpdate) {
-        document = await documentManager.findOne(id!, model, { populate, locale });
+        // check the document version
+        document = await documentManager.findOne(id!, model, { populate });
```

Comment:
I would recommend changing a bit the logic to avoid the permission copy you have to make 

pseudo code

```
if documentExists does not exist then
  throw error
if document with locale doesn't exist  then
 create and publish that new version  // runs the create permission check 1st for the new locale and then the publish permission which is more accurate
else just publish the existing one
```

Reply:
If document with locale doesn't exist but the documentExists is true I need to update the document not to create a new one, because otherwise I have different document Ids for the different locales

Reply:
Yes exactly you have to create the new locale (not a new document 👍)

Reply:
I added a new logic, trying to follow your suggestions, let me know if it works for you, thanks